tests/class.py

```diff
@@ -134,25 +134,8 @@ def test_timezone_of_datetime(self):
   def test_Collection_of_typing(self):
     self.assertOnlyIn((3, 6), self.detect("from typing import Collection"))
 
-  def test_Deque_of_typing(self):
-    self.assertOnlyIn((3, 5), self.detect("from typing import Deque"))
-    self.assertTrue(self.config.add_backport("typing"))
-    self.assertOnlyIn(((2, 7), (3, 5)), self.detect("from typing import Deque"))
-
-  def test_ContextManager_of_typing(self):
-    self.assertOnlyIn((3, 5), self.detect("from typing import ContextManager"))
-    self.assertTrue(self.config.add_backport("typing"))
-    self.assertOnlyIn(((2, 7), (3, 5)), self.detect("from typing import ContextManager"))
-
   def test_Coroutine_of_typing(self):
     self.assertOnlyIn((3, 5), self.detect("from typing import Coroutine"))
-    self.assertTrue(self.config.add_backport("typing"))
-    self.assertOnlyIn(((2, 7), (3, 5)), self.detect("from typing import Coroutine"))
-
-  def test_DefaultDict_of_typing(self):
-    self.assertOnlyIn((3, 5), self.detect("from typing import DefaultDict"))
-    self.assertTrue(self.config.add_backport("typing"))
-    self.assertOnlyIn(((2, 7), (3, 5)), self.detect("from typing import DefaultDict"))
```

Comment:
Why have you removed these and related rules from `typing`?

Reply:
Because [the source code of `typing` backport](https://github.com/python/typing/blob/master/src/typing.py) suggests that `typing.DefaultDict` is always backported. Thus we only need the module-level rule of  `typing` for this.

`typing.Coroutine` cannot be backported, because it relies on the coroutine feature which is 3.5+.

`collections.ChainMap` is 3.3+ so `typing.ChainMap` cannot be backported to 2.7. As the backport for the whole `typing` module is 2.7+ and 3.4+ as for now ([originally it even allowed 3.2+](https://github.com/python/typing/commit/02c9d79eb74b0abf05e2fd45c1127b65827fdcb4#diff-d522e03f62536e2241dbc137daaf2b4fafbf3a3771673b2e3b008e143cab9103)), the rule for `typing.ChainMap` should say 3.4+ when backported.

Then should the rule for the `typing` module become 2.7+ and 3.2+ when backported? Because on Python 3.2 you can actually install an older version of `typing` backport.

Reply:
Now it makes more sense. Hmm, yes, 2.7, 3.2 would fit better with the backported `typing` but could you then include in a comment why? Like the [link to that commit](https://github.com/python/typing/commit/02c9d79eb74b0abf05e2fd45c1127b65827fdcb4#diff-d522e03f62536e2241dbc137daaf2b4fafbf3a3771673b2e3b008e143cab9103) and a little condensed explanation from here.

Reply:
I've pushed new changes about `typing` rules. I discovered some more rules to revise and also added notes to them. Also `typing.OrderedDict` is not backported as for now (there is an [issue](https://github.com/python/typing/issues/679)).

Reply:
Looks good 👍🏻